File: tests/output/csharp/src/Models/TopModel.Sample.Restaurant.Models/CommandeItem.cs
namespace TopModel.Sample.Restaurant.Models;

public class CommandeItem : ICommandeItem
{
    public int? Id => throw new NotImplementedException();

    public DateTime? DateCommande => throw new NotImplementedException();

    public decimal? MontantTotal => throw new NotImplementedException();

    public StatutCommande? StatutCommande => throw new NotImplementedException();

    public int? ClientId => throw new NotImplementedException();

    public static ICommandeItem Create(
        int? id = null,
        DateTime? dateCommande = null,
        decimal? montantTotal = null,
        StatutCommande? statutCommande = null,
        int? clientId = null
    )
    {
        throw new NotImplementedException();
    }
}

File: tests/output/jpa/src/main/javagen/restaurant/jpa_jdbc_uuid_resttemplate/dtos/restaurant/CommandeItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_jdbc_uuid_resttemplate.dtos.restaurant;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.annotation.Generated;

import restaurant.jpa_jdbc_uuid_resttemplate.enums.restaurant.StatutCommande;

/**
 * Détail d'une commande en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface CommandeItem {

	/**
	 * Identifiant de la commande.
	 */
	Integer getId();

	/**
	 * Date et heure de la commande.
	 */
	LocalDateTime getDateCommande();

	/**
	 * Montant total de la commande.
	 */
	BigDecimal getMontantTotal();

	/**
	 * Statut de la commande.
	 */
	StatutCommande getStatutCommande();

	/**
	 * Client ayant passé la commande.
	 */
	Integer getClientId();

	/**
	 * Hydrate values of instance.
	 * @param id value to set.
	 * @param dateCommande value to set.
	 * @param montantTotal value to set.
	 * @param statutCommande value to set.
	 * @param clientId value to set.
	 */
	void hydrate(Integer id, LocalDateTime dateCommande, BigDecimal montantTotal, StatutCommande statutCommande, Integer clientId);
}
